js/src/data/actions.js

```diff
@@ -71,78 +75,91 @@ export function* fetchShippingRates() {
 	}
 }
 
-/**
- * Aggregated shipping rate.
- *
- * @typedef {Object} AggregatedShippingRate
- * @property {Array<CountryCode>} countries Array of destination country codes.
- * @property {string} currency Currency of the price.
- * @property {number} rate Shipping price.
- */
-
-/**
- * Updates or inserts given aggregated shipping rate.
- *
- * @param {AggregatedShippingRate} shippingRate
- */
-export function* upsertShippingRates( shippingRate ) {
-	const { countryCodes, currency, rate } = shippingRate;
+function* maybeDeleteShippingRates( newShippingRates, oldShippingRates ) {
```

Comment:
> the logic to get the deleteIds could go in something like `const deleteIds = getDeleteIds( newRates, oldRates)`.
> Like this deleteShippingRates is agnostic about how to get the deleteIds and it could be reused easily and the testing should be only on `getDeleteIds` that is the part doing the logic part.

That sounds good. I'll look into that in my new PR later, in relation to my comment in https://github.com/woocommerce/google-listings-and-ads/pull/1262#discussion_r817832807. 🙂 👍